src/main.rs

```diff
@@ -0,0 +1,164 @@
+use clap::Parser;
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::MerkleError,
+    Felt, Word, ONE,
+    {hash::rpo::Rpo256, merkle::TieredSmt},
+};
+use rand_utils::rand_value;
+use std::time::Instant;
+
+#[derive(Parser, Debug)]
+#[clap(
+    name = "Benchmark",
+    about = "Tiered SMT benchmark",
+    version,
+    rename_all = "kebab-case"
+)]
+pub struct BenchmarkCmd {
+    /// Size of the tree
+    #[clap(short = 's', long = "size")]
+    size: u64,
+
+    /// Run the construction benchmark
+    #[clap(short = 'c', long = "construction")]
+    construction: bool,
+
+    /// Run the insertion benchmark
+    #[clap(short = 'i', long = "insertion")]
+    insertion: bool,
+
+    /// Run the proof generation benchmark
+    #[clap(short = 'p', long = "proof-generation")]
+    proof_generation: bool,
```

Comment:
I'm not sure we need these options. Basically, to test insertion and proof generation, we'd need to do construction anyway - so, might as well report that time. And overall, benchmarking insertions/proofs is probably negligible as compared to tree construction.

So, basically, I'd probably leave size as the only option.